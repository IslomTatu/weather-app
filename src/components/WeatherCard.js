import React from 'react'
import useWeatherData from '../hooks/weatherData'
import icon from '../utils/icon'
import styles from './weathercard.module.scss'

export const WeatherCard = ({ weather, onClick }) => {
  const { weekName, date, dayValue, nightValue, info = {} } = useWeatherData(weather)

  const handleOnClick = () => {
    sessionStorage.setItem('dt', weather.dt)
    onClick(weekName.toLocaleLowerCase())
  }

  return (
    <div className={styles.card} onClick={handleOnClick}>
      <div className={styles['week-name']}>
        {weekName}
      </div>
      <div className={styles['weather-card']}>
        <div className={styles['weather-card-header']}>
          <div>
            <span>Day</span>
          </div>
          <div>
            <span>Night</span>
          </div>
        </div>
        {info?.id &&
          <div className={styles['weather-card-icon']}>
            <img src={icon[info?.id]('d')} />
            <img src={icon[info?.id]('n')} />
          </div>
        }
        <div className={styles['weather-card-info']}>
          <div className={styles['weather-card-values']}>
            <h1> {dayValue} &#8451; </h1>
            <h1> {nightValue} &#8451; </h1>
          </div>
          <div className={styles['weather-card-footer']}>
            <p className={styles['weather-card-date']}> {date} </p>
            <p> {info?.description} </p>
          </div>
        </div>
      </div>
    </div>
  )
}
