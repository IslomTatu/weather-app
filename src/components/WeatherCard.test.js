/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'

import { WeatherCard } from './WeatherCard'

describe('WeatherCard component renders testing', () => {
  it('WeatherCard renders without crashing', () => {
    expect(shallow(<WeatherCard />)).toMatchSnapshot()
  })

  it('WeatherCard renders without props', () => {
    expect(shallow(<WeatherCard />))
  })

  it('WeatherCard renders with empty weather data', () => {
    const mockProps = {
      onClick: jest.fn(),
      weather: {}
    }
    expect(shallow(<WeatherCard {...mockProps} />))
  })
})
