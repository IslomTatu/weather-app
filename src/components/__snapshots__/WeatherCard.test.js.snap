// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard component renders testing WeatherCard renders without crashing 1`] = `ShallowWrapper {}`;
